flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java

```diff
@@ -48,6 +48,8 @@
   private final ScanContext context;
 
   private transient RowDataIterator iterator;
+  private transient long currentReadCount = 0L;
+  private transient long limit;
```

Comment:
How about just using the `context.limit()`  in `reachedEnd`, rather than introducing another new transient field `limit` ?

Reply:
I deleted the field `limit `